votor/src/consensus_metrics.rs

```diff
@@ -160,69 +230,76 @@ impl ConsensusMetrics {
                 );
             }
         }
-        Ok(())
     }
 
     /// Records when a given slot started.
-    pub fn record_start_of_slot(&mut self, slot: Slot) {
-        self.start_of_slot.entry(slot).or_insert(Instant::now());
+    fn record_start_of_slot(&mut self, slot: Slot, recorded: Instant) {
+        self.start_of_slot.entry(slot).or_insert(recorded);
     }
 
     /// Performs end of epoch reporting and reset all the statistics for the subsequent epoch.
-    fn end_of_epoch_reporting(&mut self) {
+    fn end_of_epoch_reporting(&mut self, epoch: Epoch) {
         for (addr, metrics) in &self.node_metrics {
             let addr = addr.to_string();
-            datapoint_info!("votor_consensus_metrics",
+            datapoint_info!("consensus_vote_metrics",
```

Comment:
Generally I'm a fan of all the renames here to include `us`. However, I think it is typically preferable to keep changes like that in a separate PR. The metric field renames are a much less contentious and more straight forward change than the rest of this PR that adds the thread + updates callers.

If you're willing to break that into a separate PR, I'd be happy to stamp it immediately. The metric field renames are also definitely in the "votor internals" category vs. the extra thread + channels that we're discussing

Reply:
That's fair, removed all the renaming changes.